ACPs/194-streaming-asynchronous-execution/README.md

```diff
@@ -181,27 +264,26 @@ A reference implementation is still a work-in-progress. This ACP will be updated
 
 ### Worst-case transaction validity
 
-To avoid a DoS vulnerability on execution, we require an upper bound on transaction gas cost (i.e. amount $\times$ price).
-Beyond regular requirements for transaction validity (e.g. nonce, signature, etc.) we therefore introduce "worst-case cost" validity.
+To avoid a DoS vulnerability on execution, we require an upper bound on transaction gas cost (i.e. amount $\times$ price) beyond the regular requirements for transaction validity (e.g. nonce, signature, etc.). We therefore introduced "worst-case cost" validity.
 
-Consider a hypothetical situation in which all enqueued transactions use their entire gas limit during execution.
-We can prove that, for every transaction, this would result in the greatest possible:
+We can prove that, for every transaction, assuming the usage of their full gas limit during execution results in the greatest possible:
 
 1. Consumption of gas units (by definition of the gas limit); and
 2. Gas excess $x$ (and therefore gas price) at the time of execution.
 
-For a queue of transactions $Q = \\{i\\}_ {i \ge 0}$ the gas excess $x_j$ immediately prior to execution of transaction $j \in Q$ is a monotonic, non-decreasing function of the gas usage of all preceding transactions in the queue; i.e. $x_j~:=~f(\\{g_i\\}_{i<j})$.
+For a queue of blocks $Q = \\{i\\}_ {i \ge 0}$ the gas excess $x_j$ immediately prior to execution of block $j \in Q$ is a monotonic, non-decreasing function of the gas usage of all preceding blocks in the queue; i.e. $x_j~:=~f(\\{g_i\\}_{i<j})$.
 
-To see this, consider transaction $0 \le k<j$ consuming gas $g_k$.
+To see this, consider block $0 \le k<j$ consuming gas $g_k$.
 A decrease in $g_k$ reduces the immediate increase of $x$.
-Furthermore, this lowered consumption can never reduce a chunk's gas surplus and hence any decrease of $x$ is $\ge$ predicted.
-The excess, and hence gas price, for every later transaction $x_{i>k}$ is therefore reduced:
+Furthermore, this lowered consumption might further reduce $x$ at the start of executing the next block if $t^k_e < t^{k+1}_b$ and therefore $\Delta t > 0$.
+Hence any decrease of $x$ is $\ge$ predicted.
+The excess, and hence gas price, for every later block $x_{i>k}$ is therefore reduced:
 
 $$
 \downarrow g_k \implies
 \begin{cases}
-    \downarrow \frac{g \cdot (p-1)}{p} \\
-    \uparrow \frac{s}{p}
+    \downarrow \frac{g}{R} \cdot (R - T) \\
+    \downarrow \frac{g}{R}
```

Comment:
```suggestion
    \downarrow \Delta^+x \propto g_k \\
    \uparrow \Delta^-x \propto R-g_k
```

$$
\begin{cases}
    \downarrow \Delta^+x \propto g_k \\
    \uparrow \Delta^-x \propto R-g_k
\end{cases}
$$

I think $\propto$ conveys the message even if not 100% correct in the excess-reduction case.